Formula/lib/libnice-gstreamer.rb

```diff
@@ -0,0 +1,227 @@
+class LibniceGstreamer < Formula
+  desc "GStreamer Plugin for libnice"
+  homepage "https://wiki.freedesktop.org/nice/"
+  url "https://libnice.freedesktop.org/releases/libnice-0.1.22.tar.gz"
+  sha256 "a5f724cf09eae50c41a7517141d89da4a61ec9eaca32da4a0073faed5417ad7e"
+  license any_of: ["LGPL-2.1-only", "MPL-1.1"]
+
+  livecheck do
+    formula "libnice"
+  end
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => [:build, :test]
+  depends_on "glib"
+  depends_on "gnutls"
+  depends_on "gstreamer"
+  depends_on "libnice"
+
+  on_linux do
+    depends_on "intltool" => :build
+  end
+
+  # Enable building only the gstreamer plugin
+  # https://gitlab.freedesktop.org/libnice/libnice/-/merge_requests/271
+  patch :DATA
+
+  def install
+    system "meson", "setup", "build", "-Dgstreamer=enabled", "-Dgstreamer-plugin-only=true", *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    # Based on https://github.com/libnice/libnice/blob/HEAD/examples/simple-example.c
+    (testpath/"test.c").write <<~EOS
+      #include <agent.h>
+      int main(int argc, char *argv[]) {
+        NiceAgent *agent;
+        GMainLoop *gloop;
+        gloop = g_main_loop_new(NULL, FALSE);
+        // Create the nice agent
+        agent = nice_agent_new(g_main_loop_get_context (gloop),
+          NICE_COMPATIBILITY_RFC5245);
+        if (agent == NULL)
+          g_error("Failed to create agent");
+
+        g_main_loop_unref(gloop);
+        g_object_unref(agent);
+        return 0;
+      }
+    EOS
+
+    pkg_config_cflags = shell_output("pkg-config --cflags --libs nice").chomp.split
+    system ENV.cc, "test.c", *pkg_config_cflags, "-o", "test"
+    system "./test"
+  end
```

Comment:
Checking whether `gst-inspect-1.0 nicesrc` returns `0` is probably a good test.

Reply:
Presumably I would need to set some environment variables for this to work, since GStreamer wants to find all the plugins in its own prefix?

Reply:
Hm, so this should actually be built as part of the gstreamer monorepo then. Let me cook up a patch to do that.

Reply:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/6871

So, instead of creating a new `libnice-gstreamer.rb` formula, you'd patch `gstreamer.rb`